lgc/patch/FragColorExport.cpp

```diff
@@ -431,3 +484,331 @@ Value *FragColorExport::convertToInt(Value *value, bool signedness, Instruction
 }
 
 } // namespace lgc
+
+char LowerFragColorExport::ID = 0;
+
+// =====================================================================================================================
+// Create the color export pass
+ModulePass *lgc::createLowerFragColorExport() {
+  return new LowerFragColorExport();
+}
+
+// =====================================================================================================================
+// Run the lower vertex fetch pass on a module
+//
+// @param [in/out] module : Module
+bool LowerFragColorExport::runOnModule(Module &module) {
+  m_context = &module.getContext();
+  m_pipelineState = getAnalysis<PipelineStateWrapper>().getPipelineState(&module);
+  m_resUsage = m_pipelineState->getShaderResourceUsage(ShaderStageFragment);
+
+  // Gather vertex fetch calls. We can assume they're all in one function, the vertex shader.
+  // We can assume that multiple fetches of the same location, component and type have been CSEd.
```

Comment:
(Update comment.) Can we assume that each frag color location has only one export? I think we probably can, because I think the lowering pass that generates the builder calls for the exports uses a proxy variable for each one and then exports at the end. But we should check on that.

Reply:
I did a couple small tests (which I believe are already covered by existing tests), and if yo have two writes to the output variable in the spir-v, then will be merged into a single export.